sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -50,8 +50,6 @@ trait CheckAnalysis extends PredicateHelper with LookupCatalog {
 
   val DATA_TYPE_MISMATCH_ERROR = TreeNodeTag[Boolean]("dataTypeMismatchError")
 
-  val DATA_TYPE_MISMATCH_ERROR_MESSAGE = TreeNodeTag[String]("dataTypeMismatchError")
-
```

Comment:
just to make sure this isn't related to `TempResolvedColumn`, right?

Reply:
It was used to do error handling of `TempResolvedColumn`, but we don't need it now as the logic is simplified.

Reply:
> just to make sure this isn't related to `TempResolvedColumn`, right?

Yes, I added this in [XX](https://github.com/apache/spark/pull/36746). It is not needed after this pr